game_loop_entity.py

```diff
@@ -3,20 +3,67 @@
 from chess_move import ChessMove
 from lcd_interface import LCDInterface
 from chess_library import ChessLibrary
+from webserver_interface import WebServerInterface
+from pynput import keyboard
+from threading import Thread
+
+
+
+
 class GameLoopEntity():
+
     def __init__(self):
+        self.paused = False
         self.welcomed = False
         self.lcd_interface = LCDInterface()
         self.chess_library = ChessLibrary()
         self.led_interface = LedInterface()
+        self.webserver_interface = WebServerInterface()
+        self.webserver_interface.register()
+        self.thread = Thread(target=self.start_listening)
+        self.thread.start()
+
+    def start_listening(self):
+        # Collect events until released
+        with keyboard.Listener(
+                on_press=self.on_press,
+                on_release=self.on_release) as listener:
+            listener.join()
+
```

Comment:
Add a `stop_listening` function that does the following:
- Call `self.listener.stop` to stop the listener
- Call `self.thread.join()`, this will wait for the thread `self.thread` to stop